src/lro/bodyPollingStrategy.ts

```diff
@@ -0,0 +1,47 @@
+import { LROStrategy, BaseResult, LROOperationState } from "./models";
+import { OperationSpec } from "@azure/core-http";
+
+const terminalStates = ["succeeded", "failed", "canceled", "cancelled"];
+
+/**
+ * Creates a polling strategy based on BodyPolling which uses the provisioning state
+ * from the result to determine the current operation state
+ */
+export function createBodyPollingStrategy<TResult extends BaseResult>({
+  lastOperation: operation,
+  sendOperation
+}: LROOperationState<TResult>): LROStrategy<TResult> {
+  let lastOperation = { ...operation };
+  return {
+    isTerminal: () => {
+      // If provisioning state is missing, default to Success
+      const provisioningState = (
+        lastOperation.result.provisioningState ||
+        lastOperation.result.properties?.provisioningState ||
+        "succeeded"
+      ).toLowerCase();
+
+      return terminalStates.includes(provisioningState);
+    },
+    sendFinalRequest: () => {
+      // BodyPolling doesn't require a final get so return the lastOperation
+      return Promise.resolve(lastOperation);
+    },
+    poll: async () => {
+      // When doing BodyPolling, we need to poll to the original url with a
+      // GET http method
+      const pollingSpec: OperationSpec = {
+        ...lastOperation.spec,
+        httpMethod: "GET"
+      };
+
+      // Execute the polling operation
+      const result = await sendOperation(lastOperation.args, pollingSpec);
```

Comment:
So if the service doesn't support making a `GET` request with the same url/arguments this will throw an error. That seems like the right thing to do. Will that stop any further polling from happening or can you still continue to poll?

Reply:
From the implementation of the base Poller in @azure/core-lro, I think it will reject the promise so the polling will stop.

This is the where the base poller handles errors: https://github.com/Azure/azure-sdk-for-js/blob/beef206545ceca025e2114af4690a15eb8ee6bb7/sdk/core/core-lro/src/poller.ts#L323


If we want any kind of retry I a retry policy needs to be provided to coreHttp